// i18n-js/dist/require/helpers/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.timeAgoInWords = exports.strftime = exports.roundNumber = exports.propertyFlatList = exports.pluralize = exports.parseDate = exports.numberToHumanSize = exports.numberToHuman = exports.numberToDelimited = exports.lookup = exports.isSet = exports.interpolate = exports.inferType = exports.getFullScope = exports.formatNumber = exports.expandRoundMode = exports.createTranslationOptions = exports.camelCaseKeys = void 0;
var camelCaseKeys_1 = require("./camelCaseKeys");
Object.defineProperty(exports, "camelCaseKeys", { enumerable: true, get: function () { return camelCaseKeys_1.camelCaseKeys; } });
var createTranslationOptions_1 = require("./createTranslationOptions");
Object.defineProperty(exports, "createTranslationOptions", { enumerable: true, get: function () { return createTranslationOptions_1.createTranslationOptions; } });
var expandRoundMode_1 = require("./expandRoundMode");
Object.defineProperty(exports, "expandRoundMode", { enumerable: true, get: function () { return expandRoundMode_1.expandRoundMode; } });
var formatNumber_1 = require("./formatNumber");
Object.defineProperty(exports, "formatNumber", { enumerable: true, get: function () { return formatNumber_1.formatNumber; } });
var getFullScope_1 = require("./getFullScope");
Object.defineProperty(exports, "getFullScope", { enumerable: true, get: function () { return getFullScope_1.getFullScope; } });
var inferType_1 = require("./inferType");
Object.defineProperty(exports, "inferType", { enumerable: true, get: function () { return inferType_1.inferType; } });
var interpolate_1 = require("./interpolate");
Object.defineProperty(exports, "interpolate", { enumerable: true, get: function () { return interpolate_1.interpolate; } });
var isSet_1 = require("./isSet");
Object.defineProperty(exports, "isSet", { enumerable: true, get: function () { return isSet_1.isSet; } });
var lookup_1 = require("./lookup");
Object.defineProperty(exports, "lookup", { enumerable: true, get: function () { return lookup_1.lookup; } });
var numberToDelimited_1 = require("./numberToDelimited");
Object.defineProperty(exports, "numberToDelimited", { enumerable: true, get: function () { return numberToDelimited_1.numberToDelimited; } });
var numberToHuman_1 = require("./numberToHuman");
Object.defineProperty(exports, "numberToHuman", { enumerable: true, get: function () { return numberToHuman_1.numberToHuman; } });
var numberToHumanSize_1 = require("./numberToHumanSize");
Object.defineProperty(exports, "numberToHumanSize", { enumerable: true, get: function () { return numberToHumanSize_1.numberToHumanSize; } });
var parseDate_1 = require("./parseDate");
Object.defineProperty(exports, "parseDate", { enumerable: true, get: function () { return parseDate_1.parseDate; } });
var pluralize_1 = require("./pluralize");
Object.defineProperty(exports, "pluralize", { enumerable: true, get: function () { return pluralize_1.pluralize; } });
var propertyFlatList_1 = require("./propertyFlatList");
Object.defineProperty(exports, "propertyFlatList", { enumerable: true, get: function () { return propertyFlatList_1.propertyFlatList; } });
var roundNumber_1 = require("./roundNumber");
Object.defineProperty(exports, "roundNumber", { enumerable: true, get: function () { return roundNumber_1.roundNumber; } });
var strftime_1 = require("./strftime");
Object.defineProperty(exports, "strftime", { enumerable: true, get: function () { return strftime_1.strftime; } });
var timeAgoInWords_1 = require("./timeAgoInWords");
Object.defineProperty(exports, "timeAgoInWords", { enumerable: true, get: function () { return timeAgoInWords_1.timeAgoInWords; } });
