// i18n-js/dist/require/lodash.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setWith = exports.set = exports.has = exports.get = exports.flattenDeep = exports.isObject = exports.zipObject = exports.sortBy = exports.repeat = exports.snakeCase = exports.camelCase = exports.uniq = void 0;
var uniq_1 = require("lodash/uniq");
Object.defineProperty(exports, "uniq", { enumerable: true, get: function () { return __importDefault(uniq_1).default; } });
var camelCase_1 = require("lodash/camelCase");
Object.defineProperty(exports, "camelCase", { enumerable: true, get: function () { return __importDefault(camelCase_1).default; } });
var snakeCase_1 = require("lodash/snakeCase");
Object.defineProperty(exports, "snakeCase", { enumerable: true, get: function () { return __importDefault(snakeCase_1).default; } });
var repeat_1 = require("lodash/repeat");
Object.defineProperty(exports, "repeat", { enumerable: true, get: function () { return __importDefault(repeat_1).default; } });
var sortBy_1 = require("lodash/sortBy");
Object.defineProperty(exports, "sortBy", { enumerable: true, get: function () { return __importDefault(sortBy_1).default; } });
var zipObject_1 = require("lodash/zipObject");
Object.defineProperty(exports, "zipObject", { enumerable: true, get: function () { return __importDefault(zipObject_1).default; } });
var isObject_1 = require("lodash/isObject");
Object.defineProperty(exports, "isObject", { enumerable: true, get: function () { return __importDefault(isObject_1).default; } });
var flattenDeep_1 = require("lodash/flattenDeep");
Object.defineProperty(exports, "flattenDeep", { enumerable: true, get: function () { return __importDefault(flattenDeep_1).default; } });
var get_1 = require("lodash/get");
Object.defineProperty(exports, "get", { enumerable: true, get: function () { return __importDefault(get_1).default; } });
var has_1 = require("lodash/has");
Object.defineProperty(exports, "has", { enumerable: true, get: function () { return __importDefault(has_1).default; } });
var set_1 = require("lodash/set");
Object.defineProperty(exports, "set", { enumerable: true, get: function () { return __importDefault(set_1).default; } });
var setWith_1 = require("lodash/setWith");
Object.defineProperty(exports, "setWith", { enumerable: true, get: function () { return __importDefault(setWith_1).default; } });
